debug/org.eclipse.debug.ui/ui/org/eclipse/debug/internal/ui/actions/breakpoints/DisableAllBreakpointsAction.java

```diff
@@ -37,7 +37,24 @@ public class DisableAllBreakpointsAction extends AbstractRemoveAllActionDelegate
 
 	@Override
 	protected boolean isEnabled() {
-		return DebugPlugin.getDefault().getBreakpointManager().getBreakpoints().length > 0;
+		boolean allDisabled = true;
+		IBreakpointManager breakpointManager = DebugPlugin.getDefault().getBreakpointManager();
+		IBreakpoint[] breakpoints = breakpointManager.getBreakpoints();
+
+		for (IBreakpoint bp : breakpoints) {
+			try {
+				if (bp.isEnabled()) {
```

Comment:
code would be smaller and faster if you eager return false if not enabled

Reply:
will change and test it